rasa/core/training/story_conflict.py

```diff
@@ -124,7 +132,32 @@ class TrackerEventStateTuple(NamedTuple):
 
     @property
     def sliced_states_hash(self) -> int:
-        return hash(str(list(self.sliced_states)))
+        """Returns the hash of the sliced states."""
+        return hash(_as_sorted_text(self.sliced_states))
+
+
+def _as_sorted_text(obj: Any) -> Text:
+    """Returns the string of `obj` after sorting lists and dicts.
+
+    Args:
+        obj: Something made up of lists and dicts and stringifiable objects.
+
+    Returns:
+        A string representation of the object that doesn't change
+        randomly due to unsorted dicts or sets.
+    """
+    if isinstance(obj, str):
+        return obj
+    elif isinstance(obj, dict):
+        return str(
```

Comment:
can we use `json.dumps(sort_keys=True)`?

Reply:
Ah! :100: